backend/init_data/skills/dingtalk-docs/provider.py

```
@@ -0,0 +1,689 @@
+    _base_module = sys.modules.get(f"{package_name}._base")
+    if _base_module:
+        BaseSandboxTool = _base_module.BaseSandboxTool
+    else:
+        # Fallback: try to import from sandbox skill
+        try:
+            sys.path.insert(
+                0, "/workspace/1760905/Wegent/backend/init_data/skills/sandbox"
+            )
+            from _base import BaseSandboxTool
+        except ImportError:
+            raise ImportError(f"Cannot import BaseSandboxTool from any location")
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)